### PR TITLE
mobile: documenting registered extensions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low: should be comment-only
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Review by jpsim — APPROVED** [review]
+1 this is really helpful

Maybe Rafal can help audit the extensions that we might need for Lyft. I _think_ we can remove all the ones you marked as "verify with Lyft" but I'd rather be sure and tackle that as a follow-up.

**Comment by alyssawilk** [comment]
oh 100%, all changes go in follow-ups :-)

File: mobile/envoy_build_config/extension_registry.cc [hunk 1]
```diff
@@ -74,64 +74,123 @@ namespace Envoy {
...
   Extensions::UriTemplate::Match::forceRegisterUriTemplateMatcherFactory();
   Extensions::UriTemplate::Rewrite::forceRegisterUriTemplateRewriterFactory();
   Http::Matching::forceRegisterHttpRequestHeadersDataInputFactory();
   Http::Matching::forceRegisterHttpRequestTrailersDataInputFactory();
   Http::Matching::forceRegisterHttpResponseHeadersDataInputFactory();
   Http::Matching::forceRegisterHttpResponseTrailersDataInputFactory();
-  Network::Address::forceRegisterIpResolver();
-  Network::forceRegisterDefaultClientConnectionFactory();
+
+  // Envoy Mobile uses the GetAddrInfo resolver for DNS lookups by default.
```
Comment (Augustyniak) [inline mobile/envoy_build_config/extension_registry.cc:170]: May be worth calling out that it's used only on Android and that we could probably compile it out on iOS

**Review by Augustyniak — COMMENTED** [review]
+1, looks good to me. Very useful to have these comments.

**Pull request merged**